### PR TITLE
feat: IdentityService ABC, adapters, and test infrastructure

### PR DESCRIPTION
**Pull request by jamescrowley321**
## Summary
- Add `IdentityService` ABC with 21 abstract async methods returning `Result[T, IdentityError]` types
- Add `IdentityProviderAdapter` ABC with 9 abstract async methods for external IdP sync
- Add `NoOpSyncAdapter` implementation returning `Ok(None)` for all methods (test/dev use)
- Add `get_identity_service()` FastAPI dependency factory (placeholder until PostgresIdentityService in story 2.x)
- Add testcontainers-based Postgres fixtures with Alembic migrations for integration tests
- Add comprehensive unit tests for ABCs, adapters, dependency factory, and test infrastructure

## Story
Refs #142
Part of PRD 5: Canonical Identity Domain Model

## Chained PR
Based on #186 — must be merged first

## Review Findings Addressed
- Security: 0 BLOCK, 0 WARN, 4 INFO
- Blind Hunter: 3 MUST FIX (2 resolved, 1 false positive), 5 SHOULD FIX
- Edge Cases: 1 unhandled path found, 0 critical (1 DATA resolved)
- Acceptance: 4 PASS, 0 FAIL, 1 PARTIAL (intentional placeholder)
- Red Team: skipped (no auth/middleware files changed)

## Test plan
- [x] Unit tests pass (`make test-unit`)
- [x] Lint passes (`make lint`)
- [x] Independent review agents passed (blind, edge, acceptance, security)
- [ ] CI passes
- [ ] Manual verification against Descope sandbox

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: backend/tests/unit/test_identity_adapter.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+            assert "return" in hints, f"{method_name} has no return annotation"
+            ret = hints["return"]
+            origin = getattr(ret, "__origin__", None)
+            assert origin is Result, f"{method_name} return type is {ret}, not Result"
+            args = ret.__args__
+            assert args[0] is None, f"{method_name} ok type is {args[0]}, not None"
+            assert args[1] is SyncError, f"{method_name} error type is {args[1]}, not SyncError"
+
+
+class TestAdapterPartialImplementationFails:
```
Comment (jamescrowley321) [inline backend/tests/unit/test_identity_adapter.py]: This test is basically testing a language/lib feature. What is the value?
